instrumentation/opentelemetry-instrumentation-boto3sqs/src/opentelemetry/instrumentation/boto3sqs/__init__.py

```diff
@@ -0,0 +1,430 @@
+            # The method expect QueueUrl and Entries params, so if they are None, we call wrapped to receive the
+            # original exception
+            queue_name = Boto3SQSInstrumentor._extract_queue_name_from_url(
+                queue_url
+            )
+            span = self._tracer.start_span(
+                name=f"{queue_name} send",
+                kind=SpanKind.PRODUCER,
+            )
+            Boto3SQSInstrumentor._enrich_span(span, queue_name)
+            with trace.use_span(span, end_on_exit=True):
+                attributes = kwargs.pop("MessageAttributes", {})
+                propagate.inject(attributes, setter=boto3sqs_setter)
+                retval = wrapped(*args, MessageAttributes=attributes, **kwargs)
+                if message_id := retval.get("MessageId"):
+                    if span.is_recording():
+                        span.set_attribute(
+                            SpanAttributes.MESSAGING_MESSAGE_ID, message_id
+                        )
+                return retval
+
+        wrapt.wrap_function_wrapper(
+            self._sqs_class, "send_message", send_wrapper
+        )
+
+    def _wrap_send_message_batch(self) -> None:
+        def send_batch_wrapper(wrapped, instance, args, kwargs):
+            queue_url = kwargs.get("QueueUrl")
+            entries = kwargs.get("Entries")
+            # The method expect QueueUrl and Entries params, so if they are None, we call wrapped to receive the
+            # original exception
+            if (
+                context.get_value(_SUPPRESS_INSTRUMENTATION_KEY)
+                or not queue_url
+                or not entries
+            ):
+                return wrapped(*args, **kwargs)
+            queue_name = Boto3SQSInstrumentor._extract_queue_name_from_url(
+                queue_url
+            )
+            ids_to_spans: Dict[str, Span] = {}
+            for entry in entries:
+                entry_id = entry["Id"]
+                span = self._tracer.start_span(
+                    name=f"{queue_name} send",
+                    kind=SpanKind.PRODUCER,
+                )
+                ids_to_spans[entry_id] = span
+                Boto3SQSInstrumentor._enrich_span(
+                    span, queue_name, conversation_id=entry_id
+                )
+                with trace.use_span(span):
+                    if "MessageAttributes" not in entry:
+                        entry["MessageAttributes"] = {}
+                    propagate.inject(
+                        entry["MessageAttributes"], setter=boto3sqs_setter
+                    )
+            retval = wrapped(*args, **kwargs)
+            for successful_messages in retval["Successful"]:
+                message_identifier = successful_messages["Id"]
+                if message_span := ids_to_spans.get(message_identifier):
+                    if message_span.is_recording():
+                        message_span.set_attribute(
```

Comment:
same as before for message_id

Reply:
Similarly to [this case](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1081#discussion_r873561456) I add the information after because if I set all the information after, a failure will create a span with lacking information that can be provided.